udtube/modules.py

```diff
@@ -110,51 +142,6 @@ def _group_embeddings(
             ]
         ).permute(0, 2, 1)
 
-    def forward(
-        self,
-        batch: data.Batch,
-    ) -> torch.Tensor:
-        """Computes the contextual word-level encoding.
-
-        This truncates (if necessary), computes the subword encodings,
-        stacks and mean-pools the pooling layers, applies dropout, and
-        then mean-pools the subwords that make up each word.
-
-        Args:
-            batch: a data batch.
-
-        Returns:
-            A contextual word-level encoding.
-        """
-        # If something is longer than an allowed sequence, we trim it down.
-        actual_length = batch.tokens.input_ids.shape[1]
-        max_length = self.encoder.config.max_position_embeddings
-        if actual_length > max_length:
-            logging.warning(
-                "Truncating sequence from %d to %d",
-                actual_length,
-                max_length,
-            )
-            batch.tokens.input_ids = batch.tokens.input_ids[:max_length]
-            batch.tokens.attention_mask = batch.tokens.attention_mask[
-                :max_length
-            ]
-        # We move these manually rather than moving the whole batch encoding.
```

Comment:
Moving this to the collator is very smart

Reply:
It's actually going to make things faster (I think) cuz CPU can do this check in the background.